bundles/org.openhab.binding.homeconnect/src/main/java/org/openhab/binding/homeconnect/internal/client/HttpHelper.java

```diff
@@ -90,36 +90,42 @@ public static String formatJsonBody(@Nullable String jsonString) {
 
     public static String getAuthorizationHeader(OAuthClientService oAuthClientService)
             throws AuthorizationException, CommunicationException {
-        try {
-            AccessTokenResponse accessTokenResponse = oAuthClientService.getAccessTokenResponse();
-            // refresh the token if it's about to expire
-            if (accessTokenResponse != null
-                    && accessTokenResponse.isExpired(LocalDateTime.now(), OAUTH_EXPIRE_BUFFER)) {
-                LoggerFactory.getLogger(HttpHelper.class).debug("Requesting a refresh of the access token.");
-                accessTokenResponse = oAuthClientService.refreshToken();
-            }
+        synchronized ("authorizationHeader") {
```

Comment:
Are you sure this is a good idea? See https://stackoverflow.com/questions/133988/synchronizing-on-string-objects-in-java

Why not using a dedicated monitor of type `Object`?

Reply:
Thanks for the link. Changed it to a dedicated monitor object.